src/App.tsx

```diff
@@ -1,24 +1,64 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import cn from 'classnames';
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Status } from './types/Status';
+import { TodosFilter } from './components/TodosFilter/TodosFilter';
+import { Errors } from './types/Errors';
 
-const USER_ID = 0;
+const USER_ID = 7023;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredType, setFilteredType] = useState(Status.All);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+
+  const filterTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filteredType) {
+        case Status.All:
+          return true;
+        case Status.Completed:
+          return todo.completed;
+        case Status.Active:
+          return !todo.completed;
+        default:
+          return true;
+      }
+    });
+  }, [filteredType, todos]);
+
+  useEffect(() => {
+    setErrorMessage(null);
+    getTodos(USER_ID).then(todo => setTodos(todo)).catch(() => {
+      setErrorMessage(Errors.LOAD_ERROR);
+    });
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setErrorMessage(null);
+    }, 3000);
+  }, [errorMessage, setErrorMessage]);
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const leftToComplete = todos.filter(todo => !todo.completed).length;
+  const showClearButton = todos.filter(todo => todo.completed).length > 0;
+
```

Comment:
pls consider rewriting these 2 consts with useMemo, u don't need to refilter items on every render